src/main/java/org/spongepowered/api/net/ChannelStream.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.net;
+
+public interface ChannelStream {
+
+    ChannelStream writeString(String string);
+
+    String readString();
+
+    String getStringAt(int index);
+
+    ChannelStream writeInteger(int integer);
+
+    int readInteger();
+
+    int getIntegerAt(int index);
+
+    ChannelStream writeLong(long string);
+
+    long readLong();
+
+    long getLong(long index);
+
+    ChannelStream writeString(String string);
+
+    String readString();
+
+    String getStringAt(int index);
+
+
```

Comment:
I haven't quite finished the class for the sake of finishing #272. I haven't had a chance to get back to finishing this class, which of course would include floats, doubles, DataViews etc.